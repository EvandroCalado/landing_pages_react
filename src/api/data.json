[
  {
    "title": "landing_page",
    "slug": "landing-page",
    "createdAt": "2023-03-13T19:21:36.548Z",
    "updatedAt": "2023-03-14T17:43:45.879Z",
    "publishedAt": "2023-03-13T19:21:42.460Z",
    "footer": "Feito com ðŸ’• por <a href=\"https://github.com/EvandroCalado\" target=\"_blank\" >Evandro Calado</a>",
    "sections": [
      {
        "id": 1,
        "__component": "section.section-two-columns",
        "title": "JANUARY BRINGS US FIREFOX 85",
        "description": "To wrap up January, we are proud to bring you the release of Firefox 85. In this version we are bringing you support for the :focus-visible pseudo-class in CSS and associated devtools, and the complete removal of Flash support from Firefox.",
        "image": {
          "data": {
            "id": 2,
            "attributes": {
              "name": "javascript.svg",
              "alternativeText": null,
              "caption": null,
              "width": 1030,
              "height": 730,
              "formats": null,
              "hash": "javascript_200e4162bb",
              "ext": ".svg",
              "mime": "image/svg+xml",
              "size": 30.31,
              "url": "https://res.cloudinary.com/dnhsdenji/image/upload/v1678735252/javascript_200e4162bb.svg",
              "previewUrl": null,
              "provider": "cloudinary",
              "provider_metadata": {
                "public_id": "javascript_200e4162bb",
                "resource_type": "image"
              },
              "createdAt": "2023-03-13T19:20:53.913Z",
              "updatedAt": "2023-03-13T19:20:53.913Z"
            }
          }
        },
        "metadata": {
          "id": 1,
          "name": "home",
          "section_id": "home",
          "background": true
        }
      },
      {
        "id": 1,
        "__component": "section.section-content",
        "title": "NEWS COVERAGE AND SOME SURPRISES",
        "content": "The release of Apple Silicon-based Macs at the end of last year generated a flurry of news coverage and some surprises at the machineâ€™s performance. This post details some background information on the experience of porting Firefox to run natively on these CPUs.\\n\\nWeâ€™ll start with some background on the Mac transition and give an overview of Firefox internals that needed to know about the new architecture, before moving on to the concept of Universal Binaries.\\n\\nWeâ€™ll then explain how DRM/EME works on the new platform, talk about our experience with macOS Big Sur, and discuss various updater problems we had to deal with. Weâ€™ll conclude with the release and an overview of various other improvements that are in the pipeline.",
        "metadata": {
          "id": 2,
          "name": "intro",
          "section_id": "intro",
          "background": false
        }
      },
      {
        "id": 1,
        "__component": "section.section-grid-text",
        "title": "my grid",
        "description": "Uma breve descriÃ§Ã£o.",
        "text_grid": [
          {
            "id": 1,
            "title": "Teste 1",
            "description": "Lorem ipsum dolor sit, amet consectetur adipisicing elit. Debitis cum delectus molestias. Atque doloribus nobis laudantium esse ut, non commodi maxime distinctio veritatis unde, reprehenderit minus ad dolores provident maiores."
          },
          {
            "id": 3,
            "title": "Teste 2",
            "description": "Lorem ipsum dolor sit, amet consectetur adipisicing elit. Debitis cum delectus molestias. Atque doloribus nobis laudantium esse ut, non commodi maxime distinctio veritatis unde, reprehenderit minus ad dolores provident maiores."
          },
          {
            "id": 2,
            "title": "Teste 3",
            "description": "Lorem ipsum dolor sit, amet consectetur adipisicing elit. Debitis cum delectus molestias. Atque doloribus nobis laudantium esse ut, non commodi maxime distinctio veritatis unde, reprehenderit minus ad dolores provident maiores."
          }
        ],
        "metadata": {
          "id": 3,
          "name": "grid-one",
          "section_id": "grid-one",
          "background": true
        }
      },
      {
        "id": 2,
        "__component": "section.section-grid-image",
        "title": "gallery",
        "description": "Uma breve descriÃ§Ã£o.",
        "metadata": {
          "id": 5,
          "name": "gallery",
          "section_id": "gallery",
          "background": false
        },
        "image_grid": {
          "data": [
            {
              "id": 8,
              "attributes": {
                "name": "photo_1649001052863_099932ef80d1_901e8323be.jpg",
                "alternativeText": null,
                "caption": null,
                "width": 360,
                "height": 360,
                "formats": {
                  "thumbnail": {
                    "ext": ".jpg",
                    "url": "https://res.cloudinary.com/dnhsdenji/image/upload/v1678745403/thumbnail_photo_1649001052863_099932ef80d1_901e8323be_d2cc483df1.jpg",
                    "hash": "thumbnail_photo_1649001052863_099932ef80d1_901e8323be_d2cc483df1",
                    "mime": "image/jpeg",
                    "name": "thumbnail_photo_1649001052863_099932ef80d1_901e8323be.jpg",
                    "path": null,
                    "size": 5.61,
                    "width": 156,
                    "height": 156,
                    "provider_metadata": {
                      "public_id": "thumbnail_photo_1649001052863_099932ef80d1_901e8323be_d2cc483df1",
                      "resource_type": "image"
                    }
                  }
                },
                "hash": "photo_1649001052863_099932ef80d1_901e8323be_d2cc483df1",
                "ext": ".jpg",
                "mime": "image/jpeg",
                "size": 19.58,
                "url": "https://res.cloudinary.com/dnhsdenji/image/upload/v1678745403/photo_1649001052863_099932ef80d1_901e8323be_d2cc483df1.jpg",
                "previewUrl": null,
                "provider": "cloudinary",
                "provider_metadata": {
                  "public_id": "photo_1649001052863_099932ef80d1_901e8323be_d2cc483df1",
                  "resource_type": "image"
                },
                "createdAt": "2023-03-13T22:10:04.111Z",
                "updatedAt": "2023-03-14T13:38:00.889Z"
              }
            },
            {
              "id": 7,
              "attributes": {
                "name": "photo_1633843195379_b4ed419ddea2_fc5a42851e.jpg",
                "alternativeText": null,
                "caption": null,
                "width": 360,
                "height": 360,
                "formats": {
                  "thumbnail": {
                    "ext": ".jpg",
                    "url": "https://res.cloudinary.com/dnhsdenji/image/upload/v1678745403/thumbnail_photo_1633843195379_b4ed419ddea2_fc5a42851e_90f8ba2940.jpg",
                    "hash": "thumbnail_photo_1633843195379_b4ed419ddea2_fc5a42851e_90f8ba2940",
                    "mime": "image/jpeg",
                    "name": "thumbnail_photo_1633843195379_b4ed419ddea2_fc5a42851e.jpg",
                    "path": null,
                    "size": 7.93,
                    "width": 156,
                    "height": 156,
                    "provider_metadata": {
                      "public_id": "thumbnail_photo_1633843195379_b4ed419ddea2_fc5a42851e_90f8ba2940",
                      "resource_type": "image"
                    }
                  }
                },
                "hash": "photo_1633843195379_b4ed419ddea2_fc5a42851e_90f8ba2940",
                "ext": ".jpg",
                "mime": "image/jpeg",
                "size": 33.35,
                "url": "https://res.cloudinary.com/dnhsdenji/image/upload/v1678745403/photo_1633843195379_b4ed419ddea2_fc5a42851e_90f8ba2940.jpg",
                "previewUrl": null,
                "provider": "cloudinary",
                "provider_metadata": {
                  "public_id": "photo_1633843195379_b4ed419ddea2_fc5a42851e_90f8ba2940",
                  "resource_type": "image"
                },
                "createdAt": "2023-03-13T22:10:04.071Z",
                "updatedAt": "2023-03-13T22:10:04.071Z"
              }
            },
            {
              "id": 6,
              "attributes": {
                "name": "photo_1607802875758_3aaa1a149203_62d3188c12.jpg",
                "alternativeText": null,
                "caption": null,
                "width": 360,
                "height": 360,
                "formats": {
                  "thumbnail": {
                    "ext": ".jpg",
                    "url": "https://res.cloudinary.com/dnhsdenji/image/upload/v1678745403/thumbnail_photo_1607802875758_3aaa1a149203_62d3188c12_9c405b63a6.jpg",
                    "hash": "thumbnail_photo_1607802875758_3aaa1a149203_62d3188c12_9c405b63a6",
                    "mime": "image/jpeg",
                    "name": "thumbnail_photo_1607802875758_3aaa1a149203_62d3188c12.jpg",
                    "path": null,
                    "size": 8.68,
                    "width": 156,
                    "height": 156,
                    "provider_metadata": {
                      "public_id": "thumbnail_photo_1607802875758_3aaa1a149203_62d3188c12_9c405b63a6",
                      "resource_type": "image"
                    }
                  }
                },
                "hash": "photo_1607802875758_3aaa1a149203_62d3188c12_9c405b63a6",
                "ext": ".jpg",
                "mime": "image/jpeg",
                "size": 34.31,
                "url": "https://res.cloudinary.com/dnhsdenji/image/upload/v1678745403/photo_1607802875758_3aaa1a149203_62d3188c12_9c405b63a6.jpg",
                "previewUrl": null,
                "provider": "cloudinary",
                "provider_metadata": {
                  "public_id": "photo_1607802875758_3aaa1a149203_62d3188c12_9c405b63a6",
                  "resource_type": "image"
                },
                "createdAt": "2023-03-13T22:10:03.919Z",
                "updatedAt": "2023-03-13T22:10:03.919Z"
              }
            },
            {
              "id": 5,
              "attributes": {
                "name": "photo_1629997047282_41f678126175_c9c8690c82.jpg",
                "alternativeText": null,
                "caption": null,
                "width": 360,
                "height": 360,
                "formats": {
                  "thumbnail": {
                    "ext": ".jpg",
                    "url": "https://res.cloudinary.com/dnhsdenji/image/upload/v1678745403/thumbnail_photo_1629997047282_41f678126175_c9c8690c82_fda540e9c3.jpg",
                    "hash": "thumbnail_photo_1629997047282_41f678126175_c9c8690c82_fda540e9c3",
                    "mime": "image/jpeg",
                    "name": "thumbnail_photo_1629997047282_41f678126175_c9c8690c82.jpg",
                    "path": null,
                    "size": 6.59,
                    "width": 156,
                    "height": 156,
                    "provider_metadata": {
                      "public_id": "thumbnail_photo_1629997047282_41f678126175_c9c8690c82_fda540e9c3",
                      "resource_type": "image"
                    }
                  }
                },
                "hash": "photo_1629997047282_41f678126175_c9c8690c82_fda540e9c3",
                "ext": ".jpg",
                "mime": "image/jpeg",
                "size": 27.39,
                "url": "https://res.cloudinary.com/dnhsdenji/image/upload/v1678745403/photo_1629997047282_41f678126175_c9c8690c82_fda540e9c3.jpg",
                "previewUrl": null,
                "provider": "cloudinary",
                "provider_metadata": {
                  "public_id": "photo_1629997047282_41f678126175_c9c8690c82_fda540e9c3",
                  "resource_type": "image"
                },
                "createdAt": "2023-03-13T22:10:03.873Z",
                "updatedAt": "2023-03-13T22:10:03.873Z"
              }
            },
            {
              "id": 4,
              "attributes": {
                "name": "photo_1592459996257_081359053570_a9fa420eb0.jpg",
                "alternativeText": null,
                "caption": null,
                "width": 360,
                "height": 360,
                "formats": {
                  "thumbnail": {
                    "ext": ".jpg",
                    "url": "https://res.cloudinary.com/dnhsdenji/image/upload/v1678745403/thumbnail_photo_1592459996257_081359053570_a9fa420eb0_46109a0468.jpg",
                    "hash": "thumbnail_photo_1592459996257_081359053570_a9fa420eb0_46109a0468",
                    "mime": "image/jpeg",
                    "name": "thumbnail_photo_1592459996257_081359053570_a9fa420eb0.jpg",
                    "path": null,
                    "size": 3.96,
                    "width": 156,
                    "height": 156,
                    "provider_metadata": {
                      "public_id": "thumbnail_photo_1592459996257_081359053570_a9fa420eb0_46109a0468",
                      "resource_type": "image"
                    }
                  }
                },
                "hash": "photo_1592459996257_081359053570_a9fa420eb0_46109a0468",
                "ext": ".jpg",
                "mime": "image/jpeg",
                "size": 14.64,
                "url": "https://res.cloudinary.com/dnhsdenji/image/upload/v1678745403/photo_1592459996257_081359053570_a9fa420eb0_46109a0468.jpg",
                "previewUrl": null,
                "provider": "cloudinary",
                "provider_metadata": {
                  "public_id": "photo_1592459996257_081359053570_a9fa420eb0_46109a0468",
                  "resource_type": "image"
                },
                "createdAt": "2023-03-13T22:10:03.517Z",
                "updatedAt": "2023-03-13T22:10:03.517Z"
              }
            },
            {
              "id": 3,
              "attributes": {
                "name": "photo_1552666262_1f35b9080e83_f38813e26c.jpg",
                "alternativeText": null,
                "caption": null,
                "width": 360,
                "height": 360,
                "formats": {
                  "thumbnail": {
                    "ext": ".jpg",
                    "url": "https://res.cloudinary.com/dnhsdenji/image/upload/v1678745403/thumbnail_photo_1552666262_1f35b9080e83_f38813e26c_b74fa292c5.jpg",
                    "hash": "thumbnail_photo_1552666262_1f35b9080e83_f38813e26c_b74fa292c5",
                    "mime": "image/jpeg",
                    "name": "thumbnail_photo_1552666262_1f35b9080e83_f38813e26c.jpg",
                    "path": null,
                    "size": 4.18,
                    "width": 156,
                    "height": 156,
                    "provider_metadata": {
                      "public_id": "thumbnail_photo_1552666262_1f35b9080e83_f38813e26c_b74fa292c5",
                      "resource_type": "image"
                    }
                  }
                },
                "hash": "photo_1552666262_1f35b9080e83_f38813e26c_b74fa292c5",
                "ext": ".jpg",
                "mime": "image/jpeg",
                "size": 14.88,
                "url": "https://res.cloudinary.com/dnhsdenji/image/upload/v1678745402/photo_1552666262_1f35b9080e83_f38813e26c_b74fa292c5.jpg",
                "previewUrl": null,
                "provider": "cloudinary",
                "provider_metadata": {
                  "public_id": "photo_1552666262_1f35b9080e83_f38813e26c_b74fa292c5",
                  "resource_type": "image"
                },
                "createdAt": "2023-03-13T22:10:03.442Z",
                "updatedAt": "2023-03-13T22:10:03.442Z"
              }
            }
          ]
        }
      },
      {
        "id": 2,
        "__component": "section.section-grid-text",
        "title": "my grid",
        "description": "Uma breve descriÃ§Ã£o.",
        "text_grid": [
          {
            "id": 4,
            "title": "Teste 1",
            "description": "Lorem ipsum dolor sit, amet consectetur adipisicing elit. Debitis cum delectus molestias. Atque doloribus nobis laudantium esse ut, non commodi maxime distinctio veritatis unde, reprehenderit minus ad dolores provident maiores."
          },
          {
            "id": 5,
            "title": "Teste 2",
            "description": "Lorem ipsum dolor sit, amet consectetur adipisicing elit. Debitis cum delectus molestias. Atque doloribus nobis laudantium esse ut, non commodi maxime distinctio veritatis unde, reprehenderit minus ad dolores provident maiores."
          },
          {
            "id": 6,
            "title": "Teste 3",
            "description": "Lorem ipsum dolor sit, amet consectetur adipisicing elit. Debitis cum delectus molestias. Atque doloribus nobis laudantium esse ut, non commodi maxime distinctio veritatis unde, reprehenderit minus ad dolores provident maiores."
          }
        ],
        "metadata": {
          "id": 6,
          "name": "grid-two",
          "section_id": "grid-two",
          "background": true
        }
      },
      {
        "id": 1,
        "__component": "section.section-pricing",
        "title": "pricing",
        "description_one": "The release of Apple Silicon-based Macs at the end of last year generated a flurry of news coverage and some surprises at the machineâ€™s performance. This post details some background information on the experience of porting Firefox to run natively on these CPUs.",
        "description_two": "Weâ€™ll start with some background on the Mac transition and give an overview of Firefox internals that needed to know about the new architecture, before moving on to the concept of Universal Binaries.",
        "table": [
          {
            "id": 1,
            "head": "title 1",
            "body": "content1",
            "footer": "footer 1"
          },
          {
            "id": 2,
            "head": "title 2",
            "body": "content2",
            "footer": "footer 2"
          },
          {
            "id": 3,
            "head": "title 3",
            "body": "content3",
            "footer": "footer 3"
          },
          {
            "id": 4,
            "head": "title 4",
            "body": "content4",
            "footer": "footer 4"
          },
          {
            "id": 5,
            "head": "title 5",
            "body": "content5",
            "footer": "footer 5"
          }
        ],
        "metadata": {
          "id": 7,
          "name": "pricing",
          "section_id": "pricing",
          "background": false
        }
      },
      {
        "id": 1,
        "__component": "section.section-contact",
        "contact": {
          "id": 1,
          "first_name": "First Name",
          "last_name": "Last Name",
          "email": "Email",
          "message": "Message",
          "button": "Send Message"
        },
        "metadata": {
          "id": 8,
          "name": "contact",
          "section_id": "contact",
          "background": true
        }
      }
    ],
    "menu": {
      "id": 1,
      "logo_link": "#",
      "logo_text": "Logo",
      "logo": {
        "data": {
          "id": 1,
          "attributes": {
            "name": "landing_page_2_c6baa21725.svg",
            "alternativeText": null,
            "caption": null,
            "width": 47,
            "height": 9,
            "formats": null,
            "hash": "landing_page_2_c6baa21725_c39ce760eb",
            "ext": ".svg",
            "mime": "image/svg+xml",
            "size": 2.51,
            "url": "https://res.cloudinary.com/dnhsdenji/image/upload/v1678735148/landing_page_2_c6baa21725_c39ce760eb.svg",
            "previewUrl": null,
            "provider": "cloudinary",
            "provider_metadata": {
              "public_id": "landing_page_2_c6baa21725_c39ce760eb",
              "resource_type": "image"
            },
            "createdAt": "2023-03-13T19:19:09.613Z",
            "updatedAt": "2023-03-13T19:19:09.613Z"
          }
        }
      },
      "menu_links": [
        {
          "id": 1,
          "link_text": "Intro",
          "url": "#intro",
          "open_in_new_tab": false
        },
        {
          "id": 2,
          "link_text": "Grid One",
          "url": "#grid-one",
          "open_in_new_tab": false
        },
        {
          "id": 3,
          "link_text": "Gallery",
          "url": "#gallery",
          "open_in_new_tab": false
        },
        {
          "id": 4,
          "link_text": "Grid Two",
          "url": "#grid-two",
          "open_in_new_tab": false
        },
        {
          "id": 5,
          "link_text": "Pricing",
          "url": "#pricing",
          "open_in_new_tab": false
        },
        {
          "id": 6,
          "link_text": "Contact",
          "url": "#contact",
          "open_in_new_tab": false
        }
      ]
    }
  }
]
