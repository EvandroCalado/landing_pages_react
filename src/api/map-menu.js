export const mapMenu = (menu = {}) => {
  const {
    menu_links: { open_in_new_tab: newTab = false } = '',
    logo_text: text = '',
    logo_link: link = '',
    logo: { data: { attributes: { url: image = '' } = '' } = '' } = '',
    menu_links: links = [],
  } = menu;

  return {
    newTab,
    text,
    link,
    image,
    links: mapMenuLinks(links),
  };
};

export const mapMenuLinks = (links = []) => {
  return links.map((item) => {
    const {
      link_text: children = '',
      url: link = '',
      open_in_new_tab: newTab = false,
    } = item;

    return {
      children,
      link,
      newTab,
    };
  });
};
