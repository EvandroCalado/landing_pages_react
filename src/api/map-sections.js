export const mapSections = (sections = []) => {
  return sections.map((section) => {
    if (section.__component === 'section.section-two-columns') {
      return mapSectionTwoColumns(section);
    }

    if (section.__component === 'section.section-content') {
      return mapSectionContent(section);
    }

    if (section.__component === 'section.section-grid-text') {
      return mapTextGrid(section);
    }

    if (section.__component === 'section.section-grid-image') {
      return mapImageGrid(section);
    }

    return section;
  });
};

export const mapSectionTwoColumns = (section = {}) => {
  const {
    __component: component = '',
    title = '',
    description: text = '',
    image: { data: { attributes: { url: image = '' } = '' } = '' } = '',
    metadata: { background = false, section_id: sectionId = '' } = false,
  } = section;

  return {
    component,
    title,
    text,
    image,
    background,
    sectionId,
  };
};

export const mapSectionContent = (section = {}) => {
  const {
    __component: component = '',
    title = '',
    content: html = '',
    metadata: { background = false, section_id: sectionId = '' } = false,
  } = section;

  return {
    component,
    title,
    html,
    background,
    sectionId,
  };
};

export const mapTextGrid = (section = {}) => {
  const {
    __component: component = '',
    title = '',
    description = '',
    metadata: { background = false, section_id: sectionId = '' } = false,
    text_grid: grid = [],
  } = section;

  return {
    component,
    title,
    description,
    background,
    sectionId,
    grid: grid.map((text) => {
      const { title = '', description = '' } = text;

      return {
        title,
        description,
      };
    }),
  };
};

export const mapImageGrid = (section = {}) => {
  const {
    __component: component = '',
    title = '',
    description = '',
    metadata: { background = false, section_id: sectionId = '' } = false,
    image_grid,
  } = section;

  const imageGrid = image_grid[0].image_gallery.data;

  return {
    component,
    title,
    description,
    background,
    sectionId,
    grid: imageGrid.map((img) => {
      const { url: image, name: altText } = img.attributes;

      return {
        image,
        altText,
      };
    }),
  };
};
