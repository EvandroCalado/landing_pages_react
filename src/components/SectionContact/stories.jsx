import SectionContact from '.';

const SectionContactStories = {
  title: 'SectionContact',
  component: SectionContact,
  args: {
    children: 'SectionContact',
  },
  argTypes: {
    children: { type: 'string' },
  },
};

export const Template = (args) => {
  return (
    <div>
      <SectionContact {...args} />
    </div>
  );
};

export default SectionContactStories;
