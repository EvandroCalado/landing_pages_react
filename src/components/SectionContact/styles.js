import styled, { css } from 'styled-components';

export const Container = styled.div`
  ${({ theme }) => css``}
`;
export const InputsTop = styled.div`
  ${({ theme }) => css`
    width: 100%;
    display: flex;
    gap: ${theme.spacings.small};

    label {
      font-size: ${theme.font.sizes.medium};
    }

    input {
      width: 100%;
      font-size: ${theme.font.sizes.large};
      padding: 1rem 2rem;
      margin-top: ${theme.spacings.small};
      margin-bottom: ${theme.spacings.xlarge};
      border: none;
      outline: none;
      transition: all 200ms ease-in-out;

      &:focus {
        box-shadow: 0 0 10px 1px ${theme.colors.secondaryColor};
      }

      &::placeholder {
        color: ${theme.colors.mediumGray};
      }
    }
    @media ${theme.media.medium} {
      flex-direction: column;
    }
  `}
`;

export const InputsBottom = styled.div`
  ${({ theme }) => css`
    width: 100%;
    display: flex;
    flex-direction: column;

    label {
      font-size: ${theme.font.sizes.medium};
    }

    textarea {
      width: 100%;
      margin-top: ${theme.spacings.small};
      font-size: ${theme.font.sizes.large};
      padding: 1rem 2rem;
      margin-bottom: ${theme.spacings.xxlarge};
      border: none;
      outline: none;
      transition: all 200ms ease-in-out;

      &:focus {
        box-shadow: 0 0 10px 1px ${theme.colors.secondaryColor};
      }

      &::placeholder {
        color: ${theme.colors.mediumGray};
      }
    }
  `}
`;

export const Button = styled.button`
  ${({ theme }) => css`
    background-color: transparent;
    color: ${theme.colors.white};
    border: solid 5px ${theme.colors.white};
    padding: 1.5rem 2rem;
    font-size: ${theme.font.sizes.large};
    transition: all 300ms ease-in-out;
    cursor: pointer;

    :hover {
      background-color: ${theme.colors.white};
      color: ${theme.colors.primaryColor};
    }

    @media ${theme.media.medium} {
      font-size: ${theme.font.sizes.medium};
      width: 100%;
    }
  `}
`;
