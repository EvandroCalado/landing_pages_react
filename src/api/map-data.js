import { mapMenu } from './map-menu';
import { mapSections } from './map-sections';

export const mapData = (pagesData = [{}]) => {
  return pagesData.map((data) => {
    const {
      footer = '',
      slug = '',
      title = '',
      sections = [],
      menu = {},
    } = data;

    return {
      footerHtml: footer,
      slug,
      title,
      sections: mapSections(sections),
      menu: mapMenu(menu),
    };
  });
};
