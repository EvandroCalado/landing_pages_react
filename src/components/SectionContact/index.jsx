import P from 'prop-types';
import Heading from '../Heading';
import SectionBackground from '../SectionBackground';
import * as Styled from './styles';

const SectionContact = ({
  name,
  first_name,
  last_name,
  email,
  message,
  button,
  background,
  sectionId,
}) => {
  return (
    <SectionBackground background={background} sectionId={sectionId}>
      <Styled.Container>
        <Heading size="huge" as="h2" uppercase colorDark={!background}>
          {name}
        </Heading>
        <Styled.InputsTop>
          <div>
            <label htmlFor={first_name}>First name</label>
            <input type="text" name={first_name} placeholder="Your name" />
          </div>
          <div>
            <label htmlFor={last_name}>Last name</label>
            <input type="text" name={last_name} placeholder="Your last name" />
          </div>
          <div>
            <label htmlFor={email}>Email</label>
            <input type="email" name={email} placeholder="Your email" />
          </div>
        </Styled.InputsTop>
        <Styled.InputsBottom>
          <label htmlFor={message}>Message</label>
          <textarea
            name={message}
            cols="30"
            rows="10"
            placeholder="Your message"
          ></textarea>
        </Styled.InputsBottom>
        <Styled.Button>{button}</Styled.Button>
      </Styled.Container>
    </SectionBackground>
  );
};

SectionContact.propTypes = {
  first_name: P.string.isRequired,
  last_name: P.string.isRequired,
  email: P.string.isRequired,
  message: P.string.isRequired,
  button: P.string.isRequired,
  background: P.bool,
  sectionId: P.string,
};

export default SectionContact;
