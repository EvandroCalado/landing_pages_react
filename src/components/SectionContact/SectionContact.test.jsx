import SectionContact from '.';
import { screen } from '@testing-library/react';
import { renderTheme } from '../../styles/render.theme';

import dataSections from '../../api/data.json';
import { mapSectionContact } from '../../api/map-sections';
const section = dataSections[0].sections[6];

describe('<SectionContact />', () => {
  it('should render', () => {
    const data = mapSectionContact(section);
    const { container } = renderTheme(<SectionContact {...data} />);
    expect(screen.getByRole('heading')).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});
