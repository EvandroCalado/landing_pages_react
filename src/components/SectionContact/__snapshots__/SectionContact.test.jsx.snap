// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SectionContact /> should render 1`] = `
.c2 {
  color: #fff;
  font-size: 6.4rem;
  font-weight: 900;
  text-transform: uppercase;
  text-align: center;
}

.c1 {
  width: 100%;
  max-width: 120rem;
  margin: 0 auto;
  padding: 3.2rem;
}

.c0 {
  background-color: #fff;
  color: #0a1128;
  min-height: 100vh;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #0a1128;
  color: #fff;
}

.c3 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 1.6rem;
}

.c3 label {
  font-size: 2.4rem;
}

.c3 input {
  width: 100%;
  font-size: 3.2rem;
  padding: 1rem 2rem;
  margin-top: 1.6rem;
  margin-bottom: 4rem;
  border: none;
  outline: none;
  -webkit-transition: all 200ms ease-in-out;
  transition: all 200ms ease-in-out;
}

.c3 input:focus {
  box-shadow: 0 0 10px 1px #dc143c;
}

.c3 input::-webkit-input-placeholder {
  color: #ddd;
}

.c3 input::-moz-placeholder {
  color: #ddd;
}

.c3 input:-ms-input-placeholder {
  color: #ddd;
}

.c3 input::placeholder {
  color: #ddd;
}

.c4 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c4 label {
  font-size: 2.4rem;
}

.c4 textarea {
  width: 100%;
  margin-top: 1.6rem;
  font-size: 3.2rem;
  padding: 1rem 2rem;
  margin-bottom: 4.8rem;
  border: none;
  outline: none;
  -webkit-transition: all 200ms ease-in-out;
  transition: all 200ms ease-in-out;
}

.c4 textarea:focus {
  box-shadow: 0 0 10px 1px #dc143c;
}

.c4 textarea::-webkit-input-placeholder {
  color: #ddd;
}

.c4 textarea::-moz-placeholder {
  color: #ddd;
}

.c4 textarea:-ms-input-placeholder {
  color: #ddd;
}

.c4 textarea::placeholder {
  color: #ddd;
}

.c5 {
  background-color: transparent;
  color: #fff;
  border: solid 5px #fff;
  padding: 1.5rem 2rem;
  font-size: 3.2rem;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
  cursor: pointer;
}

.c5:hover {
  background-color: #fff;
  color: #0a1128;
}

@media (max-width:768px) {
  .c2 {
    font-size: 4rem;
  }
}

@media (max-width:768px) {
  .c3 {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
  }
}

@media (max-width:768px) {
  .c5 {
    font-size: 2.4rem;
    width: 100%;
  }
}

<div>
  <div
    class="c0"
    id="contact"
  >
    <div
      class="c1"
    >
      <div
        class=""
      >
        <h2
          class="c2"
        >
          contact
        </h2>
        <div
          class="c3"
        >
          <div>
            <label
              for="First Name"
            >
              First name
            </label>
            <input
              name="First Name"
              placeholder="Your name"
              type="text"
            />
          </div>
          <div>
            <label
              for="Last Name"
            >
              Last name
            </label>
            <input
              name="Last Name"
              placeholder="Your last name"
              type="text"
            />
          </div>
          <div>
            <label
              for="Email"
            >
              Email
            </label>
            <input
              name="Email"
              placeholder="Your email"
              type="email"
            />
          </div>
        </div>
        <div
          class="c4"
        >
          <label
            for="Message"
          >
            Message
          </label>
          <textarea
            cols="30"
            name="Message"
            placeholder="Your message"
            rows="10"
          />
        </div>
        <button
          class="c5"
        >
          Send Message
        </button>
      </div>
    </div>
  </div>
</div>
`;
